Add end-to-end integration tests + revert client

guenp opened the pull request:
* Add integration tests + README
* Add support for 1QBit solver
* Add config file
* Add bugfix for new `BlobDetails` model
* Revert client back to track 1 version until compatibility issues are solved (see https://github.com/microsoft/qdk-python/pull/13)

cgranade commented on azure-quantum/tests/integration/test_integration.py:
```diff
@@ -0,0 +1,163 @@
+        "QuantumMonteCarlo"
+    ]
+
+    solvers = [
+        lambda ws: SimulatedAnnealing(ws, beta_start=0),
+        lambda ws: ParallelTempering(ws, sweeps=100),
+        lambda ws: Tabu(ws, sweeps=100),
+        lambda ws: QuantumMonteCarlo(ws, trotter_number=1)
+    ]
```
Very very minor, but I'm surprised these are lambdas instead of `functools.partial`.

guenp replied:
I don't have a strong preference so happy to change it to use `partial`

cgranade commented on azure-quantum/tests/integration/test_integration.py:
```diff
@@ -0,0 +1,163 @@
+
+    # Check if IonQ solvers are enabled
+    config = get_config()
+    ionq_enabled = config["azure.quantum"].get("ionq_enabled", False)
```
These look like 1QBit solvers, not IonQ?

guenp replied:
Thanks for catching this!

cgranade commented on azure-quantum/tests/integration/README.md:
```diff
@@ -0,0 +1,14 @@
+# Integration tests
+
+These integration tests are for running end-to-end tests against a live service.
+To be able to run the tests, make sure to add a `config.ini` file to the package root (`azure-quantum`) containing your credential info in the following format:
```
Since these config files contain plaintext credentials, it may be good to proactively add `config.ini` to `.gitignore` to prevent credential leakage. GitHub should automatically scan, but it can still be helpful.

guenp replied:
Sounds good

guenp replied:
Looks like it's already in there 👍

cgranade commented on azure-quantum/azure/quantum/workspace.py:
```diff
@@ -33,27 +34,17 @@ def sdk_environment(name):
 
 ## Settings based on environment variables:
 BASE_URL_FROM_ENV = os.environ['AZURE_QUANTUM_BASEURL'] if 'AZURE_QUANTUM_BASEURL' in os.environ else None
-if  sdk_environment('dogfood'):
```
Are dogfood and canary no longer included? If so, we should make sure to remove from IQ# as well.

guenp replied:
I need to check this; will put them back in for now.

cgranade reviewed (commented):
I didn't look at the autogenerated code in detail, but at a glance it all seems reasonable. With respect to the manually written code, it looks like there's a new integration test? Is that called in any of the CI pipelines, or should it be run manually? Thanks!

guenp commented:
> Is that called in any of the CI pipelines, or should it be run manually?

See README.md; apologies for the large PR, this PR actually merges another one in as well; a more readable PR including just the integration test change is here: #14. These integration tests are still manual for now and will be included in CI/CD at a later stage.

guenp commented:
@cgranade Thanks for the detailed review! I believe I have addressed all your comments. If it's okay with you I'd like to merge this in anticipation of the release - if you have any additional comments, would you mind posting them on PR #14 to `main`?
